### ISSUE TITLE
[feature req] Selecting label text

### ISSUE THREAD

**Issue by derekdreery** [xi-editor/druid#1015]
Currently it is not possible to select text except in a text field. It would be good to also make label text optionally selectable.

**Comment by chris-morgan** [comment]
All the cases I can think of or imagine are better served by a read-only text box, rather than a selectable label. The differences are primarily appearance and keyboard accessibility.

There’s *definitely* a place for allowing selection in multi-line formatted content, but I’m not convinced there’s actually any purpose for selectable single-line text labels.

**Comment by derekdreery** [comment]
I guess I'm thinking of selection being like on the web, where everything is selectable. Whether this is desirable is an open question as far as I'm concerned: I don't know if it's desirable or not.

**Comment by cmyr** [comment]
My instinct is to agree with @chris-morgan here, but it feels mostly semantic. We want to have non-editable text that can be selected.

**Comment by PoignardAzur** [comment]
I think there's a discussion to be had on this issue.

"Text is selectable by default" vs "Text can't be selected by default" is one of the big differences between web platforms and native platforms, and I'd argue it's a case where web platforms are strictly better.

There are two class of opinions on native text selection: "I don't really see what's the interest would be" and "Oh my god, yes, I absolutely need it, I hate when I need it and I don't have it!". Actual use cases include: copying error messages to google them, copying menu strings, copying the name of an album author, copying a message someone sent me, and just idly clicking on things.

These use cases don't *strictly* require labels to be clickable by default, but I personally find it a lot more pleasant when all the text is selectable unless it's part of a button. (though I'm speaking of my desktop experience; I don't think it applies to touchscreens that well).

**Comment by cmyr** [comment]
I think this is more subtle. I think selectable text is desirable in lots of cases, but not all cases. That said, it would be nice if this were an option you could set when constructing a label.

Actual implementation is interesting, because it duplicates a lot of code from text editing, and we'd want to figure out how best to reuse that.

**Comment by derekdreery** [comment]
I agree it should be an option, but I'm not sure if it should be enabled by default or not. There are good arguments for both sides IMO.

**Comment by Maan2003** [comment]
I think it should be enabled by default but some builtin widgets like buttons should disable it by default

**Comment by james-lawrence** [comment]
I'm firmly in the camp of selectable by default. it drives me mad when I can't copy text that I need to do things with.

**Comment by jrasanen** [comment]
It is possible to style the contents of a read-only textbox?

I'd like to render markdown files, inlining the images referenced in them. And being able to select text that is quoted, pre-formatted or hyperlinked is a must have.

**Comment by derekdreery** [comment]
@jrasanen there is a project somewhere that implements a markdown renderer. I'm sorry I can't remember where it is. I'd also recommend asking questions on xi.zulipchat.com - you'll get much quicker responses! :slightly_smiling_face:
